### ISSUE TITLE
Refactor directory layout and add a Makefile to conform to modern standards

### ISSUE THREAD

**Issue by cmungall** [The-Sequence-Ontology/SO-Ontologies#462]
Refactor this repo to have a standard layout, e.g.

 * separation of source files from compiled products
 * a Makefile that controls how the compiled products are built
 * github release process
 * SOP for PRs

The easiest way to do this is to use
https://github.com/INCATools/ontology-development-kit

And create a repo from scratch. It's harder to work with an existing repo. But as a first step I recommend  @nicoleruiz or someone else on the SO team does a "dry run" where they try creating a new SO repo using ODK

Once familiar with the process, we can then either simply replace existing contents or move things around to conform

( @mikebada and @msinclair2 I recommend we also do this in MSO repo)

**Comment by mellybelly** [comment]
yes please! this would be terrific. let us know if we can help test too. @mbrush

**Comment by msinclair2** [comment]
Hi Chris,

On the MSO repo, the program MSO-SO_Generator.jar generates both the MSO and SO files automatically from a master file (also on the MSO repo), and the master file is the one that should be edited.  Ultimately we should need only one repository for both SO and MSO, I think (@mikebada and @keilbeck can correct me).  

The .jar program is a custom OWLAPI program (written by me), and somewhat cumbersome (it takes ~25 min to run).  It uses some of the ROBOT library internally, including to call the reasoner.  I think the long time is due to reasoning over MSO-SO's complex axioms to classify SO via its generically depends on relation.  I've mentioned this in other issues (e.g. ontodev/robot#368).

I could not use ROBOT when I started to write this program because it lacked functionality I needed. 
 It has greatly progressed since then.  Once ROBOT gets the ability to edit IRIs, which it does not currently have, we can switch to using a ROBOT pipeline, I think.

Your help is always greatly appreciated, of course.

**Comment by cmungall** [comment]
Thanks @msinclair2 - I made a parallel ticket in the MSO tracker. It should be easier to migrate SO to ODK since it shouldn't require any custom parts

**Comment by mikebada** [comment]
Yes, if the refactored SO is going to be automatically generated from the MSO (as we've envisioned and implemented), I would think that we'd need only one repo for both.  (I'm open to other ideas, but we thought that independently editing the MSO and SO would inevitably lead to errors and divergence.)

However @cmungall: Is this referring to the refactoring of the current public SO repo, or the refactored repo?  Or both?

**Comment by cmungall** [comment]
Yes, now I know the plan, it seems we only need one repo. I suggest preserving this repo since there is a long history of tickets here.

**Comment by davidwsant** [comment]
Hi @cmungall,

I have managed to change the structure of the repository. It previously contained several duplicated files. I have also added a Makefile, which runs with OWLTools. I believe it is far more usable under this structure. I am going to close out this issue, but leave the other two open for your suggestions as to anything else that needs to change about the structure of the repository or the process of generating the ontology files. Thanks for your help.

Thanks,

Dave Sant
